src/lib/reasoners/bitv.ml

```diff
@@ -1113,35 +1656,82 @@ module Shostak(X : ALIEN) = struct
 
   let print = Debug.print_C_ast
 
+  let nat_of_bv term bv m =
+    let mk_2pow n =
+      if n > 1 then
+        E.mk_term (Sy.Op Sy.Pow) [E.itwo; E.int (Int.to_string n)] Ty.Tint
+      else if n = 1 then E.itwo else E.ione
+    in
+    let mk_ite term n pow =
+      let nthbv = E.mk_bitv_extract n n term 1 in
+      let cond = E.mk_eq ~iff:false E.bvone nthbv in
+      E.mk_ite cond (mk_2pow pow) E.izero
+    in
+    let bvr, ctx = Canonizer.make_aux bv [] in
+    let r = Canonizer.sigma bvr in
+    let _, natl =
+      List.fold_left (
+        fun (cnt, acc) simple_t ->
+          try match simple_t with
```

Comment:
Exactly, `ite` must take a term as an argument, so I think they are. (Although there might a way to avoid it or do it in a smarter way, I'll give it a think)